Products/Zuul/routers/modelQuery.py

```diff
@@ -0,0 +1,43 @@
+##############################################################################
+#
+# Copyright (C) Zenoss, Inc. 2017, all rights reserved.
+#
+# This content is made available according to terms specified in
+# License.zenoss under the directory where your Zenoss product is installed.
+#
+##############################################################################
+
+from Products import Zuul
+from Products.Zuul.routers import TreeRouter
+from Products.ZenUtils.Ext import DirectResponse
+
+
+class ModelQueryRouter(TreeRouter):
+    """
+    A JSON/ExtDirect interface to retrieve model data from Solr
+    """
+
+    def __init__(self, context, request):
+        self.context = context
+        self.request = request
+        self.api = Zuul.getFacade('modelquery')
+        super(ModelQueryRouter, self).__init__(context, request)     
+
+
+    def _getFacade(self):
+        return self.api
+
+
+    def getDevices(self, limit=200, params=None, fields=None):
+         
+        devices, totalCount = self.api.getDevices(limit=limit, params=params, fields=fields)
+        return DirectResponse(devices=devices, totalCount=totalCount)
+
+
+    def getIndexes(self):
+
```

Comment:
It would be pretty useful if we could get indexes per object type. We could add a method in modelindex to retrieve that info instead of spitting out all fields.

Reply:
done, now getIndexes returns list of dicts where each dict represents indexes for particular object